pkg/util/bandwidth/linux.go

```diff
@@ -75,13 +76,14 @@ func (t *tcShaper) nextClassID() (int, error) {
 		if len(line) == 0 {
 			continue
 		}
-		parts := strings.Split(line, " ")
 		// expected tc line:
 		// class htb 1:1 root prio 0 rate 1000Kbit ceil 1000Kbit burst 1600b cburst 1600b
-		if len(parts) != 14 {
-			return -1, fmt.Errorf("unexpected output from tc: %s (%v)", scanner.Text(), parts)
+		classShowMatcher := regexp.MustCompile(`class\shtb\s(1:\d+)\sroot.*rate\s`)
```

Comment:
Why are you matching the "`root.*rate`" part too? Does that avoid some false positive? (If so maybe update the comment to indicate the kind of line you're trying to avoid matching too.)
Also, I'm not sure that using "`\s`" rather than just "` `" makes this any more generic/future-proof, and it makes it harder to read... (That applies throughout the patch.)
Also, you could make all of the `MustCompile` regexes be top-level variables so they only get compiled once rather than getting recompiled every time

Reply:
Since the code uses  command like `tc class add dev xxx parent 1: classid 1:x htb rate 1kbit` to add a new class where `rate` is an explicit parameter, so "`root.*rate`" ought to be matched explicitly. This is my primitive idea.
Other advice really make sense, thanks